### DISCUSSION TITLE
Can I use crazyflie-lib-python with cfclient console?

### DISCUSSION THREAD

**Discussion by cc299792458** [bitcraze/discussions#607] [Q&A] [answered]
Hi folks,

As we know, cfclient is an useful tool to print information of crazyflie. Like we can use DEBUG_PRINT() in the firmware of crazyflie, then we can print the value which we are caring about in the console of cfclient.

On the other hand, crazyflie-lib-python is a more flexible tool enable us to control crazyflie with various options, like we can use send_position_setpoint() to give a desired position and yaw to crazyflie or we can also use send_velocity_world_setpoint() to send desired velocity and yawrate to it. We can even make some customized command by adding code to crazyflie-lib-python and firmware.

So, I wonder, is there a way to combine the advantages of these two things? More specificly, can I still use crazyflie-lib-python to send command to crazyflie, at the same time, showing the feedback information of crazyflie in cfclient or anything like that?

I know there is a way that we can use log params to get some value of crazyflie. Is there a more convenient way?

Thanks,
cc299792458

**Comment by RavenLite** [comment]
I think you should add `DEBUG_PRINT()` globally at [CRTP](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/crtp.c) function for receiving packets?

**Reply by cc299792458** [reply]
Thanks for replying! Could you please explain in more details?

**Comment by krichardsson** [comment, marked as answer]
My interpretation is that you would like to get the console data in a python script. Yes, that is possible! 
Since the python client is using the python lib for communication, everything the client is doing can be done from a script as well.

Check out the [console log example](https://github.com/bitcraze/crazyflie-lib-python/blob/0.1.22/examples/console/console.py) for more details.

Logs are used to transmit the value of a variable at regular intervals.
The console log is intended for human readable logs.
There is also the memory mapping feature that can be used to transmit larger amounts of data on demand. This requires changing both firmware and python lib.
It is also possible to transmit data continuously over a CRTP port, but that requires more work, both in the firmware and the python lib.

**Reply by cc299792458** [reply]
Thanks for your elaborate instructions, sir! I will check out the website.

**Reply by cc299792458** [reply]
I find it just fit my requirement! That's so cool!
